salt/state.py

```diff
@@ -3145,13 +3145,15 @@ def __gen_opts(self, opts):
         if not isinstance(mopts, dict):
             # An error happened on the master
             opts['renderer'] = 'jinja|yaml'
-            opts['failhard'] = False
+            if 'failhard' not in opts:
+                opts['failhard'] = False
```

Comment:
```suggestion
          opts.setdefault('failhard', False)
```
This will only set failhard if it's not already set.